src/systemTest/java/com/flightstats/hub/testcase/WebhookLifecycleTest.java

```diff
@@ -0,0 +1,198 @@
+
+import static javax.ws.rs.core.Response.Status.CREATED;
+import static javax.ws.rs.core.Response.Status.OK;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+@Slf4j
+public class WebhookLifecycleTest extends BaseTest {
+
+    private static final String EMPTY_STRING = "";
+
+    private ChannelItemResourceClient channelItemResourceClient;
+    private ChannelResourceClient channelResourceClient;
+    private WebhookResourceClient webhookResourceClient;
+    private CallbackResourceClient callbackResourceClient;
+    private CallbackServer callbackServer;
+
+    private String channelName;
+    private String webhookName;
+
+    private List<String> channelItemsPosted;
+
+    @Before
+    public void setup() {
+
+        this.channelItemResourceClient = getHttpClient(ChannelItemResourceClient.class);
+        this.channelResourceClient = getHttpClient(ChannelResourceClient.class);
+        this.webhookResourceClient = getHttpClient(WebhookResourceClient.class);
+        this.callbackResourceClient = getHttpClient(CallbackResourceClient.class);
+        this.callbackServer = injector.getInstance(CallbackServer.class);
+
+        this.callbackServer.start();
+
+        this.channelName = generateRandomString();
+        this.webhookName = generateRandomString();
+    }
+
+    @Test
+    @SneakyThrows
+    public void testWebhookWithNoStartItem() {
+        final String data = "{\"webhook\": \"test1\", \"startitem\":\"false\"}";
+
+        createChannel();
+        addWebhook(buildWebhook(EMPTY_STRING));
+
+        List<String> channelItems = addItemsToChannel(data);
+        verifyWebhookCallback(channelItems);
+    }
+
+    @Test
+    @SneakyThrows
+    public void testWebhookWithStartItem() {
+        final String data = "{\"webhook\": \"test2\", \"startitem\":\"true\"}";
+
+        createChannel();
+        List<String> channelItems = addItemsToChannel(data);
+
+        addWebhook(buildWebhook(channelItems.get(4)));
+        verifyWebhookCallback(channelItems.subList(5, channelItems.size()));
+    }
+
+    @SneakyThrows
+    private void logWebhookCallbackError() {
+        log.info("Webhook name {} ", webhookName);
+
+        Call<Object> call = webhookResourceClient.getError(webhookName);
+        Response<Object> response = call.execute();
+
+        assertEquals(OK.getStatusCode(), response.code());
+        log.info("Call back errors for webhook {} {} ", webhookName, response.body());
+    }
+
+
+    private void verifyWebhookCallback(List<String> channelItems) {
+        Call<String> call = callbackResourceClient.get(callbackServer.getUrl() + "/" + this.webhookName);
+
+        try {
+            await().atMost(90, TimeUnit.SECONDS).until(() -> {
+                Response<String> response = call.clone().execute();
+                channelItemsPosted = parseResponse(response.body());
```

Comment:
It looks like `channelItemsPosted` can be a variable defined in this method rather than a member of the class.

Reply:
i know :(
Lambda function ccepts local variable only when it is final!

Reply:
Yea, you can't change a variable that's defined outside the lambda. You could probably do this though:
```channelItemsPosted.addAll(parseResponse(response.body());```
or declare channelItems via AtomicReference:
```channelItemsPosted = new AtomicReference();```
The set that in your lambda:
```channelItemsPosted.set(parseResponse(response.body());```